src/ipvs/ip_vs_proto_tcp.c

```diff
@@ -185,6 +187,8 @@ static inline int tcp_send_csum(int af, int iphdrlen, struct tcphdr *th,
         struct ipv4_hdr *iph = ip4_hdr(mbuf);
         if (rt && rt->port)
             dev = rt->port;
+        else if (conn->in_dev)
+            dev = conn->in_dev;
```

Comment:
I think it doesn't make sense to using `in_dev` when sending pakets out.

Reply:
I see. Need to add a new parameter to indentify send or receive?

Reply:
Yes. You can specify the direction (INBOUND/OUTBOUND) with a function parameter.